fairseq/data/data_utils.py

```diff
@@ -187,8 +187,52 @@ def is_batch_full(num_tokens):
 
         batch.append(idx)
 
-    # if len(batch) > 0:
-    #     yield batch
+    if len(batch) > 0:
+        yield batch
+
+
+def batch_by_size_tpu(
```

Comment:
can you just dedup functions that are `_tpu` and fold them into the original functions

Reply:
the function signatures and the code are different b/w `batch_by_ size` and `batch_by_size_tpu`. I'm going to have to respectfully decline this one.